crates/core/src/delta_datafusion/expr.rs

```diff
@@ -52,7 +52,7 @@ use crate::{DeltaResult, DeltaTableError};
 
 /// This struct is like Datafusion's MakeArray but ensures that `element` is used rather than `item
 /// as the field name within the list.
-#[derive(Debug)]
+#[derive(Debug, Hash, PartialEq, Eq)]
```

Comment:
Due to https://datafusion.apache.org/library-user-guide/upgrading.html#scalarudfimpl-aggregateudfimpl-and-windowudfimpl-traits-now-require-partialeq-eq-and-hash-traits